### PR TITLE
improve startup robustness (verify maya encoder)

### PR DESCRIPTION
**Pull request by mistafunk**
also exclude unecessary files from build

### CODE REVIEW

File: src/codec/CMakeLists.txt [hunk 1]
```diff
@@ -50,4 +50,4 @@ set_property(TARGET ${CODEC_TARGET} PROPERTY FOLDER "Serlio Maya Plugin")
 
 ### setup install target
 
-install(TARGETS ${CODEC_TARGET} DESTINATION ${INSTALL_FOLDER_PREFIX}/plug-ins/ext)
+install(TARGETS ${CODEC_TARGET} RUNTIME DESTINATION ${INSTALL_FOLDER_PREFIX}/plug-ins/ext)
```
Comment (afrischknecht) [inline src/codec/CMakeLists.txt]: Looking good. Just to be sure: the addition of RUNTIME makes `install` ignore the .lib files, right?

File: src/common.cmake [hunk 2]
```diff
@@ -21,13 +21,6 @@ elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
 endif()
 
 
-### build type
-
-if(NOT CMAKE_BUILD_TYPE)
```
Comment (afrischknecht) [inline src/common.cmake]: Fine. Would still be interested to know why it wasn't picked up.

**Comment by afrischknecht** [comment]
CI build fails on Linux. Needs to be reopened.

File: src/serlio/CMakeLists.txt [hunk 3]
```diff
@@ -100,16 +100,27 @@ srl_add_dependency_maya(${SERLIO_TARGET})
...
 set_property(TARGET ${SERLIO_TARGET} PROPERTY FOLDER "Serlio Maya Plugin")
 
+
 ### setup install target
 
-install(TARGETS ${SERLIO_TARGET} DESTINATION ${INSTALL_FOLDER_PREFIX}/plug-ins)
-install(FILES ${PRT_LIBRARIES} DESTINATION ${INSTALL_FOLDER_PREFIX}/plug-ins)
-install(FILES ${PRT_EXT_LIBRARIES} DESTINATION ${INSTALL_FOLDER_PREFIX}/plug-ins/ext)
+# serlio
+install(TARGETS ${SERLIO_TARGET} RUNTIME DESTINATION ${INSTALL_FOLDER_PREFIX}/plug-ins)
```
Comment (afrischknecht) [inline src/serlio/CMakeLists.txt]: Same issue as with codec (see above).

File: src/codec/CMakeLists.txt (hunk 1 above)
Comment (mistafunk) [reply src/codec/CMakeLists.txt]: true, why can't the cmake guys not abstract this distinction away by default??

File: src/common.cmake (hunk 2 above)
Comment (mistafunk) [reply src/common.cmake]: did not grok this fully, just saw by experiment that CMAKE_BUILD_TYPE had the value "Debug" which explained the MSI bug... but officially, the default value seems to be undefined (maybe even depending on the Generator). -> best practice is to always set it from the outside

File: src/serlio/CMakeLists.txt (hunk 3 above)
Comment (mistafunk) [reply src/serlio/CMakeLists.txt]: fixed

**Comment by mistafunk** [comment]
above issues will be fixed in a separate PR

**Pull request merged**